guiEditor/src/diagram/workbench.tsx

```diff
@@ -36,6 +36,13 @@ export const isFramePortData = (variableToCheck: any): variableToCheck is FrameP
     } else return false;
 };
 
+export enum ConstraintDirection {
+    NONE = 0,
+    SET = 1,

```

Comment:
Not fully sure what SET is, might be good to add comments on the different values